File: app/src/main/java/test/practice/android_recyclerview/RecyclerViewAdapter.java
package test.practice.android_recyclerview;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

import test.practice.R;
import test.practice.android_recyclerview.model.DataValues;

/**
 * Created by parasmani.sharma on 16/11/2017.
 */

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.MyViewHolder> {

    private final ArrayList<DataValues> list;
    private final Context context;
    private final IRecyclerViewHolderClicks l;

    public RecyclerViewAdapter(Context context, ArrayList<DataValues> list, IRecyclerViewHolderClicks l) {

        this.list = list;
        this.context = context;
        this.l = l;

    }

    public static interface IRecyclerViewHolderClicks {
        public void onClickRecyclerItem(View v, int position);
    }


    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.recycler_row, parent, false);
        MyViewHolder vh = new MyViewHolder(v);
        return vh;

    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {

        holder.txt.setText(list.get(position).getName() + "\n" + list.get(position).getEmail());

    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {

        private final TextView txt;

        public MyViewHolder(View itemView) {
            super(itemView);

            txt = (TextView) itemView.findViewById(R.id.txt);

        }
    }
}

File: app/src/main/java/test/practice/android_aaa_kotlin_app/adapter/RecyclerViewAdapter.kt
package test.practice.android_aaa_kotlin_app.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import test.practice.R
import test.practice.android_aaa_kotlin_app.model.MoviesData


/**
 * Created by parasmani.sharma on 16/11/2017.
 */
class RecyclerViewAdapter(private val context: Context, private val list: List<MoviesData>/*, private val l: IRecylerHolderClicks*/) : RecyclerView.Adapter<RecyclerViewAdapter.MyViewHolder>() {

    interface IRecylerHolderClicks {
        fun onClickRecyclerItem(v: View?, position: Int)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.recycler_row_movies, parent, false)
        return MyViewHolder(v)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {

        holder.txt.setText(list.get(position).title)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        public val txt: TextView
        public val img: ImageView

        init {
            txt = itemView.findViewById<View>(R.id.txtName) as TextView
            img = itemView.findViewById<View>(R.id.img) as ImageView
        }
    }

}
